src/Translator.php

```diff
@@ -0,0 +1,542 @@
+     */
+    private function cleanYaml()
+    {
+        foreach ($this->modulePaths as $modulePath) {
+            $name = $this->getModuleName($modulePath);
+            $this->log("Cleaning YAML sources for <info>{$name}</info>");
+            $num = 0;
+            foreach (glob($this->getLangDirectory($modulePath) . "/*.yml") as $sourceFile) {
+                $dirty = file_get_contents($sourceFile);
+                $sourceData = Yaml::parse($dirty);
+                $cleaned = Yaml::dump($sourceData, 9999, 2);
+                if ($dirty !== $cleaned) {
+                    $num++;
+                    file_put_contents($sourceFile, $cleaned);
+                }
+            }
+            $this->log("<info>{$num}</info> yml files cleaned");
+        }
+    }
+
+    /**
+     * Merge any missing keys from old json content into json files
+     */
+    private function mergeJson(): void
+    {
+        $this->log('Merging local json files');
+        foreach ($this->originalJson as $path => $contentJson) {
+            // If there are any keys in the original json that are missing now, add them back in.
+            $parsedJson = $this->jsonDecode(file_get_contents($path));
+            $contentJson = array_merge($contentJson, $parsedJson);
+            file_put_contents($path, $this->jsonEncode($contentJson));
+        }
+        $this->log('Finished merging ' . count($this->originalJson) . ' json files');
+    }
+
+    /**
+     * Run text collector on the given modules
+     */
+    private function collectStrings(): void
+    {
+        $this->log("Running i18nTextCollectorTask");
+        $modulesNames = [];
+        foreach ($this->modulePaths as $modulePath) {
+            $modulesNames[] = $this->getModuleName($modulePath);
+        }
+        $module = urlencode(implode(',', $modulesNames));
+        $site = rtrim($this->txSite, '/');
+        $this->exec("wget $site/dev/tasks/i18nTextCollectorTask?flush=all&merge=1&module=$module");
+    }
+
+    /**
+     * Push source updates to transifex
+     */
+    private function transifexPushSource(): void
+    {
+        $this->log('Pushing updated sources to transifex');
+        if ($this->isDevMode) {
+            $this->log('Not pushing to transifex because TX_DEV_MODE is enabled');
+            return;
+        }
+        foreach ($this->modulePaths as $modulePath) {
+            $this->exec('tx push -s', $modulePath);
+        }
+    }
+
+    /**
+     * Commit changes for all modules
+     */
+    private function gitCommitPushAndPullRequest(): void
+    {
+        $this->log('Committing translations to git');
+        foreach ($this->modulePaths as $modulePath) {
+            // Get endpoint
+            $remote = $this->exec('git config --get remote.origin.url', $modulePath);
+            $remote = str_replace('git@github.com:', 'https://github.com/', $remote);
+            if (!preg_match('#^https://github.com/([^/]+)/(.+?)$#', $remote, $matches)) {
```

Comment:
```suggestion
            if (!preg_match('#^(https://github\.com/|git@github\.com:)([^/]+)/(.+?)(\.git)?$#', $remote, $matches)) {
```
This way we don't need the confusing `str_replace` above or the redundant `preg_replace` below.

Reply:
Updated